### ISSUE TITLE
Segmentation fault while running train.py

### ISSUE THREAD

**Issue by ZijianWu1121** [notmahi/dobb-e#4]
HI, I run the train.py following the imitation-in-homes  README

I download all the datasets and change the path in env_vars.yaml. Other than that, all configurations are default.
but when I run the command 
python train.py --config-name=finetune

The logs show:

[2024-02-06 15:48:42,360][__main__][INFO] - Working in ./dobb-e/imitation-in-homes
[2024-02-06 15:48:43,065][__main__][INFO] - Setting up dataloaders.
Filtering trajectories: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 81/81 [00:00<00:00, 132814.16it/s]
./dobb-e/imitation-in-homes/dataloaders/utils.py:283: UserWarning: No trajectories were found matching the specified criteria. Make sure you have the correct dataset root and trajectory roots, and the correct include/exclude criteria.
  warnings.warn(
Filtering trajectories: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 81/81 [00:00<00:00, 155700.56it/s]
[2024-02-06 15:48:43,941][timm.models._builder][INFO] - Loading pretrained weights from Hugging Face hub (notmahi/dobb-e)
[2024-02-06 15:48:44,306][timm.models._hub][INFO] - [notmahi/dobb-e] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
Using static action normalization
act_mean: tensor([ 7.2950e-05,  5.2102e-03,  9.5942e-04, -1.9970e-04, -1.5271e-03,
         3.3574e-04,  0.0000e+00])
act_std: tensor([0.0144, 0.0221, 0.0110, 0.0158, 0.0465, 0.0253, 1.0000])
Fatal Python error: Segmentation fault

I use faulthandler and find that the error occurred in
self.model = self.model.to(self.device)

**Comment by notmahi** [comment]
I think the most important error message is here:

`./dobb-e/imitation-in-homes/dataloaders/utils.py:283: UserWarning: No trajectories were found matching the specified criteria. Make sure you have the correct dataset root and trajectory roots, and the correct include/exclude criteria.`

Could you double check your dataset folder and config?

**Comment by wzjscut** [comment]
> I think the most important error message is here:
> 
> `./dobb-e/imitation-in-homes/dataloaders/utils.py:283: UserWarning: No trajectories were found matching the specified criteria. Make sure you have the correct dataset root and trajectory roots, and the correct include/exclude criteria.`
> 
> Could you double check your dataset folder and config?


thanks for the reply.
i just follow the setting in env_vars.yaml.sample.

home_ssl_data_root is the path of iphone data download from google drive.
finetune_task_data_root is the same.

but how about xxx_original_root?  i can't find the file r3d_files.txt so i set the same path like home_ssl_data_root and finetune_task_data_root.

**Comment by ZijianWu1121** [comment]
> I think the most important error message is here:
> 
> `./dobb-e/imitation-in-homes/dataloaders/utils.py:283: UserWarning: No trajectories were found matching the specified criteria. Make sure you have the correct dataset root and trajectory roots, and the correct include/exclude criteria.`
> 
> Could you double check your dataset folder and config?


Thanks for your reply.
I fix the problem of dataset folder but it shows the same log.


[2024-02-07 13:29:06,511][__main__][INFO] - Working in /home/ubuntu/dobb-e/imitation-in-homes
[2024-02-07 13:29:07,212][__main__][INFO] - Setting up dataloaders.
Filtering trajectories: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 81/81 [00:00<00:00, 4959.47it/s]
Filtering trajectories: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 81/81 [00:00<00:00, 26786.93it/s]
[2024-02-07 13:29:08,433][timm.models._builder][INFO] - Loading pretrained weights from Hugging Face hub (notmahi/dobb-e)
[2024-02-07 13:29:08,790][timm.models._hub][INFO] - [notmahi/dobb-e] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
Using static action normalization
act_mean: tensor([ 7.2950e-05,  5.2102e-03,  9.5942e-04, -1.9970e-04, -1.5271e-03,
         3.3574e-04,  0.0000e+00])
act_std: tensor([0.0144, 0.0221, 0.0110, 0.0158, 0.0465, 0.0253, 1.0000])
Fatal Python error: Segmentation fault

Thread 0x00007f01669f7700 (most recent call first):
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 316 in wait
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 581 in wait
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/tqdm/_monitor.py", line 60 in run
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 937 in _bootstrap

Thread 0x00007f01b26c3700 (most recent call first):
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 316 in wait
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 581 in wait
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/sentry_sdk/metrics.py", line 489 in _flush_loop
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 917 in run
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 980 in _bootstrap_inner
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/threading.py", line 937 in _bootstrap

Current thread 0x00007f02774e7740 (most recent call first):
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/cuda/__init__.py", line 302 in _lazy_init
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1150 in convert
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 825 in _apply
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 802 in _apply
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 802 in _apply
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 802 in _apply
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1152 in to
  File "/home/ubuntu/dobb-e/imitation-in-homes/train.py", line 103 in init_everything
  File "/home/ubuntu/dobb-e/imitation-in-homes/train.py", line 52 in __init__
  File "/home/ubuntu/dobb-e/imitation-in-homes/train.py", line 410 in main
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/core/utils.py", line 186 in run_job
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/_internal/hydra.py", line 119 in run
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/_internal/utils.py", line 458 in <lambda>
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/_internal/utils.py", line 220 in run_and_report
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/_internal/utils.py", line 457 in _run_app
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/_internal/utils.py", line 394 in _run_hydra
  File "/home/ubuntu/anaconda3/envs/dobbe/lib/python3.9/site-packages/hydra/main.py", line 94 in decorated_main
  File "/home/ubuntu/dobb-e/imitation-in-homes/train.py", line 415 in <module>

**Comment by notmahi** [comment]
Sorry, this information is insufficient to try and debug your issue. Can you give more information with your RAM, OS, GPU details?

Also, please try the extended logging suggested here and let us know what the error message is there: https://blog.richard.do/2018/03/18/how-to-debug-segmentation-fault-in-python/

**Comment by ZijianWu1121** [comment]
> Sorry, this information is insufficient to try and debug your issue. Can you give more information with your RAM, OS, GPU details?
> 
> Also, please try the extended logging suggested here and let us know what the error message is there: https://blog.richard.do/2018/03/18/how-to-debug-segmentation-fault-in-python/


I have use -Xfaulthandler to print the logs.
My device is NVIDIA GeForce RTX 3090(24GB) and the driver is NVIDIA-SMI 535.154.05 
My system is Ubuntu20.04
PyTorch version is 2.2.0
Cuda version is 12.1
python is 3.9.18

**Comment by ZijianWu1121** [comment]
> Sorry, this information is insufficient to try and debug your issue. Can you give more information with your RAM, OS, GPU details?

Is it possible that the performance of my GPU is insufficient, and how should I modify the parameters？

**Comment by notmahi** [comment]
I recommend you try the suggestion from @haritheja-e  given below first, and if that doesn't work, try the suggestion in the rest of the answer.

---

Hi, we are still deliberating what the issue might be. In the meantime, would you mind sending us your processed dataset, zipped? You can find my email on the Dobb-E paper or my website, https://mahis.life

**Comment by haritheja-e** [comment]
Hi, I've seen a similar segmentation fault occur once before when training and it had to do with the order of imports, specifically with torch. 

Could you try moving the torch import to the first line of train.py? 

Changing the first 11 lines in train.py from this: 
```
import logging
import shutil
from datetime import timedelta
from itertools import chain
from os import environ
from pathlib import Path
from typing import Iterable, Tuple

import decord
import hydra
import torch
```

To this: 
```
import torch
import logging
import shutil
from datetime import timedelta
from itertools import chain
from os import environ
from pathlib import Path
from typing import Iterable, Tuple

import decord
import hydra
```

Please let us know whether this resolves the segfault error, or whether it still persists.

**Comment by ZijianWu1121** [comment]
> Hi, I've seen a similar segmentation fault occur once before when training and it had to do with the order of imports, specifically with torch.
> 
> Could you try moving the torch import to the first line of train.py?
> 
> Changing the first 11 lines in train.py from this:
> 
> ```
> import logging
> import shutil
> from datetime import timedelta
> from itertools import chain
> from os import environ
> from pathlib import Path
> from typing import Iterable, Tuple
> 
> import decord
> import hydra
> import torch
> ```
> 
> To this:
> 
> ```
> import torch
> import logging
> import shutil
> from datetime import timedelta
> from itertools import chain
> from os import environ
> from pathlib import Path
> from typing import Iterable, Tuple
> 
> import decord
> import hydra
> ```
> 
> Please let us know whether this resolves the segfault error, or whether it still persists.

IT DOES WORK !
Thank you so much !!!
Could you explain why that is?

**Comment by notmahi** [comment]
> > Please let us know whether this resolves the segfault error, or whether it still persists.
> 
> IT DOES WORK ! Thank you so much !!! Could you explain why that is?

I'm sorry, we don't understand this very well either! But it seems like it's above our pay grade, since people were having this issue seven years ago.

Here's a discussion on pytorch forums: https://discuss.pytorch.org/t/odd-segmentation-fault-if-torch-is-imported/854
